handyrl/train.py

```diff
@@ -414,8 +414,49 @@ def run(self):
         print('finished training')
 
 
+def preprocess_config(args):
+    default_config = {
+        'train_args': {
+            'turn_based_training': True,
+            'observation': False,
+            'gamma': 0.8,
+            'forward_steps': 16,
+            'compress_steps': 4,
+            'entropy_regularization': 1.0e-1,
+            'entropy_regularization_decay': 0.1,
+            'update_episodes': 200,
+            'batch_size': 128,
+            'minimum_episodes': 400,
+            'maximum_episodes': 100000,
+            'epochs': -1,
+            'num_batchers': 2,
+            'eval_rate': 0.1,
+            'worker': {
+                'num_parallel': 6,
+            },
+            'lambda': 0.7,
+            'policy_target': 'TD',
+            'value_target': 'TD',
+            'seed': 0,
+            'restart_epoch': 0,
+        },
+    }
+
+    def recursive_update(dct, merge_dct):
+        for k, v in merge_dct.items():
+            if k in dct and isinstance(dct[k], dict) and isinstance(v, dict):
+                recursive_update(dct[k], v)
+            elif not k in dct:
```

Comment:
Please let me know if I am wrong. Shouldn't the output of the below code be 10?
```
$ preprocess_config({'train_args': {'epochs': 10}})['train_args']['epochs']
-1
```

Isn't this line is just `else:` or others?
```
else:
    dct[k] = v
```

Reply:
Thank you for your precise advice!
Replacing `elif not k in dct` with 'else' seems to work.